### DISCUSSION TITLE
Registry URL is handled inconsistent, depending on custom registry support

### DISCUSSION THREAD

**Discussion by mikaello** [renovatebot/renovate#17807] [Suggest an Idea]
A datasource can support custom registries, when it does support custom registries they will [be massaged](https://github.com/renovatebot/renovate/blob/1c81eaf24dd33e9e211fa742c86a54675019d749/lib/modules/datasource/index.ts#L186-L188) (i.e. trailing slashes removed), when it does not support custom registries it will not be massaged (trailing slashes preserved).

The relevant function is this:

https://github.com/renovatebot/renovate/blob/1c81eaf24dd33e9e211fa742c86a54675019d749/lib/modules/datasource/index.ts#L190-L231

Notice the massaging at the bottom (line 230), but not on line 212-214.

**Why does this matter?**

* Seemingly inconsistent, harder to see how a registry URL is handled
* If converting a datasource that previously did not support custom registries to start supporting it, the package URL resolve logic needs to be changed (probably small fix, adding a `/` between `registryUrl` and the package name) and snapshot tests will need an update.

**Context**

enabling custom registries for golang-version (https://github.com/renovatebot/renovate/discussions/17787)

**Possible solution**

* Ignore this (and keep removing slashes for custom registries, but not for default registries), and just fix the package name resolving when changing from not supporting to supporting custom registries
* Never remove slashes
* Remove slashes in both cases, example: https://github.com/renovatebot/renovate/compare/main...mikaello:renovate:fix-datasource-registry-url-inconsintency?expand=1

**Comment by rarkins** [comment]
Use https://github.com/renovatebot/renovate/blob/ea5806887b81b9800b941656b139e0333062f8fb/lib/util/url.ts#L8-L10
